telemetry/internal/appinsights/inmemorychannel.go

```diff
@@ -80,8 +82,16 @@ func newInMemoryChannel(endpointUrl string, batchSize int, batchInterval time.Du
 	return channel
 }
 
-func (channel *inMemoryChannel) logf(format string, args ...any) {
-	channel.errorLog.Printf(format, args...)
+func (channel *inMemoryChannel) error(err error) {
+	channel.errorsMu.Lock()
+	defer channel.errorsMu.Unlock()
+	channel.errors = append(channel.errors, err)
```

Comment:
Does this have any limit? Otherwise it can be a pseudo-memory-leak for a long-running application (like my use case).

Reply:
I logged in to mark this PR as not ready for review just for this reason 😓. Thanks!

Reply:
Ok, I repurposed this PR with a completely different approach (see new PR description).